### PR TITLE
Fix STA learning rate

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Use detector learning rate for STA instead of backbone learning rate

This fixes a bug where we mistakenly applied the backbone learning rate to the STA in LTDETR. As the STA is randomly initialized during training the backbone learning rate is too low and leads to slow weight updates. Instead, this PR applies the default detector learning rate instead which follows the DEIMv2 implementation.

## How has it been tested?

* Manually

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 1]
```diff
@@ -432,8 +448,10 @@ def get_optimizer(
                 }
             )
 
-        detector_params = list(self.model.encoder.parameters()) + list(
-            self.model.decoder.parameters()
+        detector_params = (
+            connector_params
+            + list(self.model.encoder.parameters())
+            + list(self.model.decoder.parameters())
         )
```
[the remark this answers is not in the document]
Comment (guarin) [reply src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:457]: This is addressed in https://github.com/lightly-ai/lightly-train/pull/647

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 2]
```diff
@@ -432,8 +448,10 @@ def get_optimizer(
...
         detector_params_wd = [p for p in detector_params if p not in params_no_wd]
         detector_params_no_wd = [p for p in detector_params if p in params_no_wd]
```
[the remark this answers is not in the document]
Comment (guarin) [reply src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:459]: Addressed in https://github.com/lightly-ai/lightly-train/pull/647

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 3]
```diff
@@ -430,8 +446,10 @@ def get_optimizer(
...
 
-        detector_params = list(self.model.encoder.parameters()) + list(
-            self.model.decoder.parameters()
+        detector_params = (
+            connector_params
+            + list(self.model.encoder.parameters())
+            + list(self.model.decoder.parameters())
         )
         detector_params_wd = [p for p in detector_params if p not in params_no_wd]
         detector_params_no_wd = [p for p in detector_params if p in params_no_wd]
```
[the remark this answers is not in the document]
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:457]: Addressed in https://github.com/lightly-ai/lightly-train/pull/647

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 4]
```diff
@@ -430,8 +446,10 @@ def get_optimizer(
                 }
             )
 
-        detector_params = list(self.model.encoder.parameters()) + list(
-            self.model.decoder.parameters()
+        detector_params = (
+            connector_params
+            + list(self.model.encoder.parameters())
```
[the remark this answers is not in the document]
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:453]: Addressed in https://github.com/lightly-ai/lightly-train/pull/647

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 5]
```diff
@@ -430,8 +446,10 @@ def get_optimizer(
...
+            + list(self.model.decoder.parameters())
```
Comment (yutong-xiang-97) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:454]: From Codex:

When `detector_weight_decay` differs from `backbone_weight_decay` (which is the default here: 0.0 vs 1e-4), prepending `connector_params` to `detector_params` changes their regularization as well as their LR. That means the newly moved wrapper weights like `backbone.convs.*.weight` now train with no decay.

So in DEIMv2, the adapter conv weights like `sta.*.weight` and `convs.*.weight` get the full/base LR, but they also keep the default weight decay. Only norm/bias parameters get zero decay. That is why your current change does not match DEIMv2 exactly: it matches the LR split, but it also moves those adapter weights onto `detector_weight_decay` instead of leaving them on the default/backbone decay.

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 6]
```diff
@@ -432,8 +448,10 @@ def get_optimizer(
                 }
             )
 
-        detector_params = list(self.model.encoder.parameters()) + list(
-            self.model.decoder.parameters()
+        detector_params = (
+            connector_params
```
Comment (yutong-xiang-97) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:454]: From Codex:

This has the same regularization regression on the DINOv3 path: `connector_params` are now optimized with the detector group's `weight_decay`, so the STA/projection conv weights lose the backbone decay whenever `detector_weight_decay != backbone_weight_decay` (again true by default: 0.0 vs 1e-4). As written, the patch changes both LR and weight decay for `backbone.sta.*.weight` / `backbone.convs.*.weight`, not just the intended learning-rate split.

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py (hunk 5 above)
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:454]: This is addressed in https://github.com/lightly-ai/lightly-train/pull/647 no?

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py (hunk 6 above)
Comment (guarin) [reply src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:454]: This is addressed in https://github.com/lightly-ai/lightly-train/pull/647 no?

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py (hunk 5 above)
Comment (yutong-xiang-97) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:454]: Apparently it was a bad review order :)
